backend/src/neuroagent/app/config.py

```diff
@@ -4,14 +4,30 @@
 from typing import Literal, Optional
 
 from dotenv import dotenv_values
-from pydantic import BaseModel, ConfigDict, SecretStr
+from pydantic import BaseModel, ConfigDict, SecretStr, model_validator
 from pydantic_settings import BaseSettings, SettingsConfigDict
 
 
 class SettingsAgent(BaseModel):
     """Agent setting."""
 
-    model: Literal["simple", "multi"] = "simple"
+    composition: Literal["simple", "multi"] = "multi"
```

Comment:
Discussed already but this is probably the biggest "red flag" I am seeing in here. From the conceptual point of view. I am almost sure that if we merge this then we will just end up maintaining one of the two.